diff --git a/node_modules/@types/openseadragon/index.d.ts b/node_modules/@types/openseadragon/index.d.ts
index 8b45a56..417acb5 100644
--- a/node_modules/@types/openseadragon/index.d.ts
+++ b/node_modules/@types/openseadragon/index.d.ts
@@ -350,14 +350,15 @@ declare namespace OpenSeadragon {
         showSequenceControl?: boolean | undefined;
         sequenceControlAnchor?: ControlAnchor | undefined;
         navPrevNextWrap?: boolean | undefined;
-        zoomInButton?: string | undefined;
-        zoomOutButton?: string | undefined;
-        homeButton?: string | undefined;
-        fullPageButton?: string | undefined;
-        rotateLeftButton?: string | undefined;
-        rotateRightButton?: string | undefined;
-        previousButton?: string | undefined;
-        nextButton?: string | undefined;
+        zoomInButton?: string | undefined | Element;
+        zoomOutButton?: string | undefined | Element;
+        homeButton?: string | undefined | Element;
+        fullPageButton?: string | undefined | Element;
+        rotateLeftButton?: string | undefined | Element;
+        rotateRightButton?: string | undefined | Element;
+        previousButton?: string | undefined | Element;
+        nextButton?: string | undefined | Element;
+        drawer?: 'webgl' | 'canvas' | 'html';
         sequenceMode?: boolean | undefined;
         /**
          * If sequenceMode is true, display this page initially.
